Assets/Scripts/SS3D/Systems/Chat/ChatTab.cs

```diff
@@ -0,0 +1,196 @@
+                    {
+                        index = i;
+                    }
+                }
+            }
+            // Otherwise figure out how far left
+            else
+            {
+                for (int i = tabRow.childCount - 1; i >= 0; i--)
+                {
+                    RectTransform child = (RectTransform) tabRow.GetChild(i);
+                    float pos = child.position.x;
+                    // consider old pos of tab since it is being held currently
+                    if (child == (RectTransform)chatTab.transform)
+                    {
+                        pos = chatTab._oldPos.x;
+                    }
+
+                    if (pos + (child.rect.width / 2) > mousePos.x)
+                    {
+                        index = i;
+                    }
+                }
+            }
+            
+            // Put tab back where it was
+            if (index == transform.GetSiblingIndex())
+            {
+                transform.position = chatTab._oldPos;
+            }
+            // Otherwise move tab over to new spot
+            else
+            {
+                transform.SetSiblingIndex(index);
+            }
+        }
+
+        public void OnEndDrag(PointerEventData eventData)
+        {
+            InGameChatWindow inGameChatWindow = eventData.hovered
+                .FirstOrDefault(x => x.GetComponentInParent<InGameChatWindow>() != null)?
+                .GetComponentInParent<InGameChatWindow>();
+            if (inGameChatWindow != null)
+            {
+                // Check if dropping on the same window
+                if (inGameChatWindow == _inGameChatWindow)
+                {
+                    PlaceTab(this, _inGameChatWindow, eventData.position);
+                    _inGameChatWindow.SelectTab(gameObject);
+                }
+                else
+                {
+                    ChatTab chatTab = inGameChatWindow.AddTab(data);
+                    PlaceTab(chatTab, inGameChatWindow, eventData.position);
+                    Button a = _inGameChatWindow.GetNextTabButton(gameObject);
+                    gameObject.Dispose(false);
+
+                    if (a == null)
+                    {
+                        _inGameChatWindow.gameObject.Dispose(false);
+                    }
+                    else
+                    {
+                        _inGameChatWindow.SelectTab(a.gameObject);
+                    }
+
+                    return;
+                }
+            }
+            else
+            {
+                // Create a new chat window as long as there are multiple tabs
+                if (_inGameChatWindow.GetTabCount() > 1)
+                {
+                    InGameChatWindow newInGameChatWindow = Instantiate(inGameChatWindowPrefab).GetComponent<InGameChatWindow>();
+                    newInGameChatWindow.transform.SetParent(_inGameChatWindow.transform.parent);
```

Comment:
You can cache the Transform component to not access it each time. Accessing a Transform component from gameObject.transform is slow due to it being code that is run in cpp, its faster to cache.